src/signals/incident-management/components/CalendarInput/index.js

```diff
@@ -1,33 +1,60 @@
-import React, { Fragment } from 'react';
+import React, { Fragment, useCallback, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import DatePicker, { registerLocale } from 'react-datepicker';
 import { dateToString } from 'shared/services/date-utils';
 import nl from 'date-fns/locale/nl';
 import CustomInput from './CustomInput';
+import 'react-datepicker/dist/react-datepicker.css';
+import styled from 'styled-components';
 registerLocale('nl', nl);
 
-const CalendarInput = ({ id, label, name, onSelect, selectedDate }) => (
-  <Fragment>
-    <DatePicker
-      autoComplete="off"
-      customInput={<CustomInput label={label} id={id} />}
-      dateFormat="dd-MM-yyyy"
-      id={id}
-      locale="nl"
-      onChange={onSelect}
-      selected={selectedDate}
-    />
-
-    {selectedDate && (
-      <input
-        value={dateToString(selectedDate)}
+/** This input is focused after a date is selected to enable form submit functionality for this control*/
+const SelectedDateInput = styled.input`
+  border: 0;
+  margin: 0;
+  padding: 0;
+  width: 0;
+  overflow: hidden;
+`;
+
+const CalendarInput = ({ id, label, name, onSelect, selectedDate }) => {
+  const inputRef = useRef(null);
+
+  const handleChange = useCallback(
+    event => {
+      onSelect(event);
```

Comment:
Can't you just pass `onSelect` to the `DatePicker`?

Reply:
Not if I want to enable submit on Enter. `onSelect` was passed to `DatePicker` in the original version. The aim is now to make it work with the keyboard.

Reply:
This seems to be just a passthrough function though. How does it help?

Reply:
Ik get it now! I thought you meant to link it to `onSelect` event van the `DatePicker`! I'll file a chore PR asap